src/main/kotlin/org/rust/lang/core/psi/impl/mixin/RustLifetimeParamElementImplMixin.kt

```diff
@@ -0,0 +1,13 @@
+package org.rust.lang.core.psi.impl.mixin
+
+import com.intellij.lang.ASTNode
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RustLifetimeParamElement
+import org.rust.lang.core.psi.RustTokenElementTypes
+import org.rust.lang.core.psi.impl.RustNamedElementImpl
+
+abstract class RustLifetimeParamElementImplMixin(node: ASTNode) : RustNamedElementImpl(node), RustLifetimeParamElement {
+
+    override fun getNameIdentifier(): PsiElement? = findChildByType(RustTokenElementTypes.LIFETIME)
```

Comment:
To expand on this a bit, here's an interesting entanglement of the base and derived class happens!

The leaf class of the hierarchy is `RustLifetimeParamElementImpl`, and we want to define `getNameIdentifier` there. We don't have direct control over the leaf, so we define a base class mixin and implement the method there. 

However, we employ `getLifetime` which is itself defined in the leaf class! In the ancestor, we are able to use the method defined in the descendant because both ancestor and the descendant share the same interface `RustLifetimeParamElement`!